tidy3d/components/medium.py

```diff
@@ -2551,8 +2551,9 @@ def _derivative_field_cmp(
         eps_data: PermittivityDataset,
         dim: str,
     ) -> np.ndarray:
-        coords_interp = {key: val for key, val in eps_data.coords.items() if len(val) > 1}
-        dims_sum = {dim for dim in eps_data.coords.keys() if dim not in coords_interp}
+        coords_interp = {key: eps_data.coords[key] for key in "xyz"}
+        coords_interp = {key: val for key, val in coords_interp.items() if len(val) > 1}
```

Comment:
Maybe `coords_interp = {key: val for key, val in eps_data.coords.items() if key in "xyz" and len(val) > 1}`?

Reply:
I actually like mine better since this one breaks over two lines.. sometimes I need to modify these comprehensions and have trouble following when they split over several lines.

Reply:
yeah fair